comp/core/workloadmeta/collectors/internal/ecs/ecs.go

```diff
@@ -121,13 +151,26 @@ func (c *collector) GetTargetCatalog() workloadmeta.AgentType {
 func (c *collector) parseTasks(ctx context.Context, tasks []v1.Task) []workloadmeta.CollectorEvent {
 	events := []workloadmeta.CollectorEvent{}
 	seen := make(map[workloadmeta.EntityID]struct{})
+	// get task ARNs to fetch from the metadata v4 API
+	taskArns := c.getTaskArnsToFetch(tasks)
 
 	for _, task := range tasks {
 		// We only want to collect tasks without a STOPPED status.
 		if task.KnownStatus == "STOPPED" {
 			continue
 		}
 
+		// If the orchestrator explorer ecs collection is enabled, collector will query the v4 task endpoint for each task
+		// and store task and container in the store.
+		if c.v4TaskEnabled {
```

Comment:
We are checking the value of `c.v4TaskEnabled` for every task. Is there a possibility for `v4TaskEnabled` to change between iterations?

Reply:
It's impossible as this value is set at the start of workloadmeta. I updated this part to split them into 2 functions.I hope this change makes the it clearer than before.

https://github.com/DataDog/datadog-agent/blob/882d6e4354906f23c0124aa0d908eed0e5bcbd4a/comp/core/workloadmeta/collectors/internal/ecs/ecs.go#L135-L139